internal/plugin/client.go

```diff
@@ -106,57 +112,74 @@ func NewAccountClient(ctx context.Context, config *configEntry) (*accountClientC
 }
 
 func (clientCtx *projectClientContext) GenerateToken(projectName string, projectRoleName string, expiresIn time.Duration) (*projectToken, error) {
-	id := uuid.New().String()
-	createTokenRequest := &project.ProjectTokenCreateRequest{
-		Project:   projectName,
-		Role:      projectRoleName,
-		ExpiresIn: toDurationSeconds(expiresIn),
-		Id:        id,
-	}
+	retries := 0
+	var response *project.ProjectTokenResponse
+	var err error
+
+	for retries < totalRetries {
+		id := uuid.New().String()
+		time.Sleep(retryWaitSeconds[retries] * time.Second)
+		createTokenRequest := &project.ProjectTokenCreateRequest{
+			Project:   projectName,
+			Role:      projectRoleName,
+			ExpiresIn: toDurationSeconds(expiresIn),
+			Id:        id,
+		}
+
+		projectClient := clientCtx.client
+		response, err = projectClient.CreateToken(clientCtx.clientContext, createTokenRequest)
+
+		if err == nil {
```

Comment:
small nitpick but in golang its typica to check if the error != nil and error out if true instead of checking if it is not nil and putting the logic in the brackets

Reply:
+1

Reply:
Let me know if you guys think otherwise but we have to check that the err is nil. When the err is nil, we want to break out of the loop. If we only return failure when err is not nil, when there is no error, it will go through the loop again and do a retry even when we don't want to. 

Happy to discuss further if you see an alternative.

Reply:
what about err != nil then you can increment retries and `continue`

Reply:
couldn't you just add a break so that it doesn't continue loop again when no error?

Reply:
The if-statement to check the break condition would be on `if err == nil ` though so we would still need to check for this condition. This solution would require us to do two checks (technically 3 since there's an && needed and this check should already be encompassed by the for loop condition). I.e.:

```
for retries < totalRetries {
 logic
  if err != nil && retries < totalRetries -1 {
    return error
  } else if err == nil {
    break
  }
logic
}
```

Happy to implement it this way if you still think it's better but if we're checking for `if err == nil` anyway in this solution, and also since the for loop is already handling the `retries < totalRetries` seems like it makes sense to not make the changes. Let me know if you think otherwise.